Add LLM.txt guide for AI assistants

raifdmueller opened the pull request:
## Overview

This PR addresses issue #1493 by adding an LLM.txt file to the repository. This file is designed specifically to help Large Language Models (LLMs) like ChatGPT, Claude, and others provide more accurate and up-to-date information about docToolchain.

## What's Included

The LLM.txt file provides:

1. A comprehensive overview of what docToolchain is
2. Current installation instructions focusing on the `dtcw` wrapper
3. Basic configuration guidance
4. Common commands and task categories
5. Solutions to common errors
6. Clarification of common misconceptions that LLMs often have about docToolchain
7. Example workflows

## Why This Matters

Many users now rely on AI assistants to help them use tools like docToolchain. However, these AI systems often provide outdated or incorrect information because:
- Their training data may include older versions of docToolchain
- They may rely on outdated information sources like docs-as-co.de
- They lack a single, authoritative source of knowledge about docToolchain

This file addresses these issues by providing a dedicated resource formatted for LLM consumption that can be referenced in the README to improve the accuracy of AI-assisted support.

## Next Steps

If merged, we should:
1. Add a reference to this file in the README.md
2. Keep it updated with each release
3. Consider replacing it with a more structured format (like LLM.md) in the future

Closes #1493

raifdmueller committed:
Add LLM.txt guide for AI assistants

rdmueller reviewed (changes requested):
That is a great start! 
- since the file is written in markdown, let's rename it from .txt to .md
- I guess we can add some more detailed information. Please read the docs and tutorials and add all needed knowledge.

raifdmueller commented:
Hi @rdmueller,

Thank you for creating issue #1493 which prompted this PR. I've structured the LLM.txt file to be comprehensive yet focused on the most critical information that AI assistants should know about docToolchain.

Some key points about this implementation:

1. **Format Choice**: I chose plain text (.txt) format as it's universally compatible and easily readable by all LLMs. We could consider Markdown (.md) in the future for better formatting.

2. **Content Structure**: The content progresses from general overview to specific details, with particular emphasis on correcting common misconceptions.

3. **Warning Sections**: I've included clear "IMPORTANT" markers for sections that address the most frequently misunderstood aspects of docToolchain.

4. **Example-Rich**: The file contains concrete examples that LLMs can reference directly when providing guidance.

5. **Versioning Info**: The file includes version information at the top to help LLMs understand when the content might become outdated.

If merged, it would be beneficial to add a reference to this file in the README.md to increase its visibility to both users and LLMs. I'd be happy to submit a follow-up PR for that change if desired.

I'm open to any suggestions for improvements to make this more effective for AI assistants!

raifdmueller committed:
Rename LLM.txt to LLM.md

raifdmueller committed:
Remove LLM.txt as it is replaced by LLM.md

raifdmueller committed:
Enhance LLM.md with additional advanced features and examples

raifdmueller commented:
Based on your feedback @rdmueller, I've made the following changes:

1. Renamed the file from LLM.txt to LLM.md to reflect the Markdown format
2. Added several new sections with detailed information from the documentation:
   - Advanced Features section with details on diagram integration, external systems, document customization, and automation
   - Environment Configuration section with examples of environment variables
   - More detailed Advanced Configuration Examples for Jira, Confluence, HTML, and OpenAPI
3. Added more common commands and task descriptions
4. Added more common errors and solutions
5. Added more misconceptions to help LLMs provide accurate information

The enhanced file should now be even more helpful for AI assistants when guiding users on docToolchain usage. Let me know if you'd like further improvements or have any additional suggestions!

rdmueller commented on LLM.md:
```diff
@@ -0,0 +1,398 @@
+## Key Features
+
+- **AsciiDoc Support**: Primarily uses AsciiDoc markup language, which offers more built-in features than Markdown
+- **Arc42 Integration**: Built with strong support for the arc42 template for software architecture documentation
```
arc42 is always written lowercase

raifdmueller replied:
Fixed in my upcoming commit. I'll change Arc42 to lowercase arc42 throughout the document.

rdmueller commented on LLM.md:
````diff
@@ -0,0 +1,398 @@
+
+```bash
+# Option 1: Run docToolchain in a container (recommended)
+./dtcw tasks --group doctoolchain
````
I think we should remove this first option

raifdmueller replied:
I'll remove this option in my upcoming commit and keep only the local installation approach.

rdmueller commented on LLM.md:
```diff
@@ -0,0 +1,398 @@
+./dtcw tasks --group doctoolchain
+
+# Option 2: Install locally
+./dtcw local install doctoolchain
```
we should also add
./dtcw local install java
before install doctoolchain

raifdmueller replied:
I'll add `./dtcw local install java` step before the docToolchain installation in my upcoming commit.

rdmueller commented on LLM.md:
````diff
@@ -0,0 +1,398 @@
+```bash
+./dtcw downloadTemplate
+```
+
````
should we add a hint that it can also be run through docker instead of local?

raifdmueller replied:
I'll add detailed information about Docker usage in a separate section right after the installation instructions in my upcoming commit.

rdmueller commented on LLM.md:
```diff
@@ -0,0 +1,398 @@
+./dtcw convertToDocx
+
+# Start the autobuild process (watches for changes)
+./dtcw autorebuild
```
is it really called "autorebuild"? And I think this option is outdated

raifdmueller replied:
I'll remove the autorebuild command in my upcoming commit and update the automation section to refer to file system watchers instead.

rdmueller commented on LLM.md:
```diff
@@ -0,0 +1,398 @@
+   **Solution**: Check Confluence credentials in the config file and ensure you're using an API token instead of password
+
+5. **Error**: Java version issues
+   **Solution**: docToolchain supports Java 8-14; check your Java version
```
I think we dropped support for java 8 and 11 in the latest changes. Please verify this

raifdmueller replied:
I'll update the Java version information to show that docToolchain requires Java 17 now, and will remove the references to Java 8 and 11.

rdmueller commented on LLM.md:
```diff
@@ -0,0 +1,398 @@
+   **Solution**: docToolchain supports Java 8-14; check your Java version
+
+6. **Error**: Diagram generation fails
+   **Solution**: Ensure required dependencies are installed (Graphviz, PlantUML, etc.)
```
or use kroki to generate diagrams

raifdmueller replied:
I'll add Kroki as a diagram generation option in my upcoming commit.

rdmueller reviewed (changes requested):
looks great. I added some comments.

raifdmueller commented:
Thank you for the thorough review, @rdmueller. I'll address all your feedback:

1. ✅ Change `arc42` to lowercase throughout the document
2. ✅ Remove "Option 1" from installation methods and only keep the local installation approach
3. ✅ Add `./dtcw local install java` step before installing doctoolchain
4. ✅ Add information about running docToolchain through Docker
5. ✅ Remove the outdated `autorebuild` command
6. ✅ Update Java version information to reflect current support status
7. ✅ Add Kroki as an option for diagram generation

I'll update the PR with these changes shortly.

raifdmueller commented:
Based on the review feedback, I'll be making the following changes in separate commits to avoid potential issues with large file updates:

1. First, I'll change Arc42 to lowercase arc42
2. Then I'll modify the installation steps to remove Option 1 and add ./dtcw local install java
3. Next, I'll add Docker information and update the automation section
4. Finally, I'll update the Java version information and add Kroki to the diagram section

I'll start with these changes now.

raifdmueller committed:
Update LLM.md based on review feedback

- Changed Arc42 to lowercase arc42
- Removed Option 1 from installation methods
- Added ./dtcw local install java step
- Added Docker information
- Removed outdated autorebuild command
- Updated Java version information
- Added Kroki as a diagram generation option
- Updated automation section to use file watchers instead of autorebuild

raifdmueller commented:
@rdmueller I've made all the requested changes in my latest commit:

1. Changed Arc42 to lowercase arc42 throughout the document
2. Removed Option 1 from installation methods and only kept the local installation approach
3. Added `./dtcw local install java` step before installing doctoolchain
4. Added a dedicated Docker Usage section right after the installation instructions
5. Removed the outdated autorebuild command from the Common Commands section
6. Updated the Java version information to specify that docToolchain requires Java 17 now
7. Added Kroki as an option for diagram generation
8. Updated the automation section to use file system watchers instead of autorebuild

The PR is now ready for your review again. Let me know if any further changes are needed!

rdmueller reviewed (approved):
great. thanx for the good work.
